### PR TITLE
chore: migrates to eslint

### PR DESCRIPTION
**Pull request by dzucconi**
Figured it'd make sense to pull this out of my working branch and PR it separately.

### CODE REVIEW

File: packages/palette/tslint.json [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-        "types": {
-          "visibilities": ["exported"]
-        },
-        "variables": {
-          "visibilities": ["exported"]
-        }
-      }
-    }
-  }
-}
```
Comment (dzucconi) [inline packages/palette/tslint.json:38]: So I didn't migrate all of these obviously. I tried using a tool to do it automatically but it requires us to update tslint to even use it.

If there's anything that stands out as being critical here lemme know?

Comment (damassi) [reply packages/palette/tslint.json:38]: Nah, it's all good. (Those migration tools are crazy)

**Review by damassi — APPROVED** [review]
Thank you!

**Comment by artsyit** [comment]
:rocket: PR was released in [`@artsy/palette@17.7.1`](https://github.com/artsy/palette/releases/tag/%40artsy/palette%4017.7.1) :rocket:

**Pull request merged**